Build failure: nct6687d with linuxPackages_latest

### Steps To Reproduce

Steps to reproduce the behavior:
1. nixos-rebuild with 
```
  boot.kernelModules = [ "nct6687" "i2c-dev" ];
  boot.extraModulePackages = with config.boot.kernelPackages; [ nct6687d ];
  boot.kernelPackages = pkgs.linuxPackages_latest;
```

### Build log

```
error: builder for '/nix/store/0dll4a4kvayqd4lyhclnswnqg395sqa6-nct6687d-0-unstable-2024-02-23.drv' failed with exit code 2;
       last 10 log lines:
       >       |
       > /build/source/nct6687.c:1171:19: error: initialization of 'void (*)(struct platform_device *)' from incompatible pointer type 'int (*)(struct platform_device *)' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
       >  1171 |         .remove = nct6687_remove,
       >       |                   ^~~~~~~~~~~~~~
       > /build/source/nct6687.c:1171:19: note: (near initialization for 'nct6687_driver.<anonymous>.remove')
       > cc1: some warnings being treated as errors
       > make[2]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/scripts/Makefile.build:244: /build/source/nct6687.o] Error 1
       > make[1]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:1926: /build/source] Error 2
       > make: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:224: __sub-make] Error 2
       > make: Leaving directory '/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build'
       For full logs, run 'nix log /nix/store/0dll4a4kvayqd4lyhclnswnqg395sqa6-nct6687d-0-unstable-2024-02-23.drv'.
error: 1 dependencies of derivation '/nix/store/hvpika8wh7wah8y76i7fn4z182hxbsjj-linux-6.11-modules.drv' failed to build
error (ignored): error: cannot unlink '/tmp/nix-build-nvidia-x11-560.35.03-6.11.drv-0/build/NVIDIA-Linux-x86_64-560.35.03/32': Directory not empty
error: 1 dependencies of derivation '/nix/store/8v1pnrv39kdmf431ss363n1m1swq2pyb-nixos-system-lacerta-24.11.20240916.99dc878.drv' failed to build

```

### Additional context
Just broke a few days ago. Right now I'm trying out Nyx's Cachyos Kernel, but I tried to change to pkgs.linuxPackages_latest. I have not tried to build it on vanilla kernel, as any rebuild fails right now due to this error. Something similar also happened with my framework laptop, but that seems to be resolved #342261

### Notify maintainers
@Atemu 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Yes, its cachyos kernel, I added ``` boot.kernelPackages = pkgs.linuxPackages_latest;``` to my config though and the issue persists.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.10-cachyos, NixOS, 24.11 (Vicuna), 24.11.20240913.345c263`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - nixpkgs: `/nix/store/l3amk5lsakpc93him5kry24kax23sn4h-source`

```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/os-specific/linux/nct6687d/default.nix
+++ b/pkgs/os-specific/linux/nct6687d/default.nix
@@ -7,13 +7,13 @@
 
 stdenv.mkDerivation rec {
   pname = "nct6687d";
-  version = "0-unstable-2024-02-23";
+  version = "0-unstable-2024-09-02";
 
   src = fetchFromGitHub {
     owner = "Fred78290";
     repo = "nct6687d";
-    rev = "0ee35ed9541bde22fe219305d1647b51ed010c5e";
-    hash = "sha256-g81U+ngnsOslBDCQP51uDDyHPpCv9T/j+KmFUAQfz/M=";
+    rev = "f45b72a6ca335e20f8a2df5ee5d4e130916150d0";
+    hash = "sha256-2sAL6aXUafmm2DKIhqEoTSPZVmzu/lnTkpPznJndX3o=";
   };
 
   setSourceRoot = ''

